homeassistant/components/xiaomi_miio/fan.py

```diff
@@ -990,14 +1022,27 @@ def __init__(self, name, device, model, unique_id):
         if self._model in [MODEL_AIRHUMIDIFIER_CA1, MODEL_AIRHUMIDIFIER_CB1]:
             self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_CA_AND_CB
             self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_CA_AND_CB
+            self._led_brightness_enum = AirhumidifierLedBrightness
+            self._led_brightness_default = 2
+            self._operation_mode_enum = AirhumidifierOperationMode
             self._speed_list = [
                 mode.name
                 for mode in AirhumidifierOperationMode
                 if mode is not AirhumidifierOperationMode.Strong
             ]
+        elif self._model in [MODEL_AIRHUMIDIFIER_JSQ001]:
+            self._device_features = FEATURE_FLAGS_AIRHUMIDIFIER_JSQ
+            self._available_attributes = AVAILABLE_ATTRIBUTES_AIRHUMIDIFIER_JSQ
+            self._led_brightness_enum = AirhumidifierJsqLedBrightness
+            self._led_brightness_default = 0
+            self._operation_mode_enum = AirhumidifierJsqOperationMode
+            self._speed_list = [mode.name for mode in AirhumidifierJsqOperationMode]
```

Comment:
We only allow the modes in the fan base entity integration:
https://github.com/home-assistant/core/blob/cedf7e3945ff387215018e6a4368c099fd9d8990/homeassistant/components/fan/__init__.py#L35-L38

Reply:
Same issue as in https://github.com/home-assistant/core/pull/31729#pullrequestreview-387488433